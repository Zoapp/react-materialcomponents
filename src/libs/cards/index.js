/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import CardActions from "./actions";
import Card from "./card";
import CardMedia from "./media";
import CardText from "./text";
import CardPrimary from "./primary";
import CardTitle from "./title";
import CardSubtitle from "./subtitle";

export {
  Card,
  CardActions,
  CardMedia,
  CardText,
  CardPrimary,
  CardTitle,
  CardSubtitle,
};
