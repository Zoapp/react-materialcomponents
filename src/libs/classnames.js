/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
  Rewritten to ES7 by Mik BRY
*/
const ClassNames = (args) => {
  const classes = [];
  args.forEach((arg) => {
    if (arg) {
      const argType = typeof arg;
      if (argType === "string" || argType === "number") {
        classes.push(arg);
      } else if (Array.isArray(arg) && arg.length) {
        const inner = ClassNames(arg);
        if (inner) {
          classes.push(inner);
        }
      } else if (argType === "object") {
        Object.keys(arg).forEach((key) => {
          if (arg[key]) {
            classes.push(key);
          }
        });
      }
    }
  });
  return classes.join(" ");
};

export default ClassNames;
