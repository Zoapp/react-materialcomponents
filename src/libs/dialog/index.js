/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Dialog from "./dialog";
import DialogBody from "./body";
import DialogFooter from "./footer";
import DialogHeader from "./header";

export { Dialog, DialogBody, DialogFooter, DialogHeader };
