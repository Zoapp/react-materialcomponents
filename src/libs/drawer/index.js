import Drawer from "./drawer";
import DrawerContent from "./content";
import DrawerHeader from "./header";

export { Drawer, DrawerContent, DrawerHeader };
