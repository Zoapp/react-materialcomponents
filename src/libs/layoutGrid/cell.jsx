/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Zrmc from "../";

/**
 * mdc-layout-grid__cell
 * See:
 * https://material.io/components/web/catalog/layout-grid/
 *
 */

const MDC_GRIDCELL = "mdc-layout-grid__cell";

const Cell = ({ children, span, spanDevice, order, align, ...props }) => {
  let classes = MDC_GRIDCELL;
  if (span > 0 && span < 13) {
    classes += ` mdc-layout-grid__cell--span-${span}`;
  } else if (spanDevice) {
    if (
      spanDevice.desktop &&
      spanDevice.desktop > 0 &&
      spanDevice.desktop < 13
    ) {
      classes += ` mdc-layout-grid__cell--span-${spanDevice.desktop}-desktop`;
    }
    if (spanDevice.tablet && spanDevice.tablet > 0 && spanDevice.tablet < 13) {
      classes += ` mdc-layout-grid__cell--span-${spanDevice.tablet}-tablet`;
    }
    if (spanDevice.phone && spanDevice.phone > 0 && spanDevice.phone < 13) {
      classes += ` mdc-layout-grid__cell--span-${spanDevice.phone}-phone`;
    }
  }
  if (order > 0 && order < 13) {
    classes += ` mdc-layout-grid__cell--order-${order}`;
  }
  if (align) {
    classes += ` mdc-layout-grid__cell--align-${align}`;
  }
  return Zrmc.render(<div className={classes}>{children}</div>, props);
};

Cell.defaultProps = {
  mdcElement: MDC_GRIDCELL,
  children: null,
  span: 0,
  spanDevice: null,
  order: 0,
  align: null,
};

Cell.propTypes = {
  mdcElement: PropTypes.string,
  children: PropTypes.node,
  span: PropTypes.number,
  spanDevice: PropTypes.shape({
    desktop: PropTypes.number,
    tablet: PropTypes.number,
    phone: PropTypes.number,
  }),
  order: PropTypes.number,
  align: PropTypes.oneOf(["top", "middle", "bottom"]),
};

export default Cell;
