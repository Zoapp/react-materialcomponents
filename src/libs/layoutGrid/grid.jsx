/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import PropTypes from "prop-types";
import Rmdc from "../";

/*
mdc-layout-grid
See:
https://material.io/components/web/catalog/layout-grid/

TODO:
- Ripple effect
- Mixins
*/
const Grid = ({
  children, className, ...props
}) => {
  let classes = "mdc-layout-grid";
  if (className) {
    classes += ` ${className}`;
  }
  return Rmdc.render(<div className={classes} {...props}>{children}</div>, props);
};

Grid.defaultProps = {
  children: null,
  className: null,
};

Grid.propTypes = {
// React component props
  children: PropTypes.node,
  className: PropTypes.string,
};

export default Grid;
