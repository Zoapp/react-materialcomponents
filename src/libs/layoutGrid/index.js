/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Grid from "./grid";
import Inner from "./inner";
import Cell from "./cell";

export { Grid, Inner, Cell };
