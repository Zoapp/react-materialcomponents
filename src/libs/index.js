/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import Rmdc from "./rmdc";
import Button from "./components/button";
import Icon from "./components/icon";
import Content from "./components/content";
import Checkbox from "./components/checkbox";
import Fab from "./components/fab";
import TextField from "./components/textfield";
import Radio from "./components/radio";
import FormField from "./components/formField";
import Select from "./components/select";
import { Menu, MenuAnchor, MenuItem, MenuDivider } from "./menu";
import Snackbar from "./components/snackbar";
import LinearProgress from "./components/linearProgress";
import Switch from "./components/switch";
import Slider from "./components/slider";
import IconToggle from "./components/iconToggle";
import { Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon } from "./toolbar";
import { Drawer, DrawerContent, DrawerHeader } from "./drawer";
import { List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader } from "./list";
import { Grid, Inner, Cell } from "./layoutGrid";
import { Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle } from "./cards";
import { Tabbar, Tab } from "./tab";
import { GridList, Tile } from "./gridList";
import { Dialog, DialogHeader, DialogBody, DialogFooter } from "./dialog";

export {
  Button, Content, Fab, TextField, Checkbox, Radio, FormField, Select, Snackbar,
  LinearProgress, Switch, Slider, IconToggle, Icon,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerContent, DrawerHeader,
  List, ListItem, ListItemMeta, ListDivider, ListGroup, ListGroupSubheader,
  Grid, Inner, Cell,
  Card, CardActions, CardMedia, CardText, CardPrimary, CardTitle, CardSubtitle,
  Tabbar, Tab,
  GridList, Tile,
  Dialog, DialogHeader, DialogBody, DialogFooter,
  Menu, MenuAnchor, MenuItem, MenuDivider,
};

export default Rmdc;
