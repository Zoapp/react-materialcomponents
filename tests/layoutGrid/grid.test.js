import React from "react";
import renderer from "react-test-renderer";
import { Grid } from "libs/layoutGrid";

describe("layoutGrid/Grid", () => {
  it("can have children", () => {
    const tree = renderer.create(<Grid>Children are here!</Grid>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a fixed columns width", () => {
    const tree = renderer.create(<Grid fixedColumnWidth />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can take differents margins on a specify device", () => {
    const tree = renderer
      .create(<Grid spanDevice={{ phone: "100" }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can take differents gutter on a specify device", () => {
    const tree = renderer.create(<Grid gutter={{ desktop: "200" }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can have a max width", () => {
    const tree = renderer.create(<Grid maxWidth="100" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can be aligned", () => {
    const tree = renderer.create(<Grid align="right" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
