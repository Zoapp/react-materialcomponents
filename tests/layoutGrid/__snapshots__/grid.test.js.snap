// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layoutGrid/Grid can be aligned 1`] = `
<div
  className="mdc-layout-grid mdc-layout-grid--align-right"
/>
`;

exports[`layoutGrid/Grid can have a fixed columns width 1`] = `
<div
  className="mdc-layout-gridmdc-layout-grid--fixed-column-width"
/>
`;

exports[`layoutGrid/Grid can have a max width 1`] = `
<div
  className="mdc-layout-grid"
  style={
    Object {
      "maxWidth": "100",
    }
  }
/>
`;

exports[`layoutGrid/Grid can have children 1`] = `
<div
  className="mdc-layout-grid"
>
  Children are here!
</div>
`;

exports[`layoutGrid/Grid can take differents gutter on a specify device 1`] = `
<div
  className="mdc-layout-grid"
  style={
    Object {
      "--mdc-layout-grid-gutter-desktop": "200",
    }
  }
/>
`;

exports[`layoutGrid/Grid can take differents margins on a specify device 1`] = `
<div
  className="mdc-layout-grid"
/>
`;
